lib/offline/OfflineStorage_SQLite.cpp

```diff
@@ -17,6 +17,9 @@ namespace MAT_NS_BEGIN {
 
     constexpr static size_t kBlockSize = 8192;
 
+    std::mutex OfflineStorage_SQLite::m_initAndShutdownLock;
+    int OfflineStorage_SQLite::m_instanceCount = 0;
```

Comment:
Nit: it's conventional (but really not necessary) to make reference counts either size_t or uint64_t. Perhaps because we all remember the trauma (no, we don't ALL remember it) from designs where someone said "oh, we don't need that sign bit," or in the case of IBM's System 360 and 370, we don't need the high byte, who could ever need more than a 24 bit address. Don't ask me, I just work here. I believe that the probability that the SDK will need more than 2 billion SQLite-consuming-things is vanishingly small, so I am absolutely ok with the signed instance count here.

One can also make this work with std::shared_ptr, but I think your design is more straightforward.

Reply:
Good point. In this case I was slightly more worried about accidental underflows, but I'm not expecting either (underflow or 2 bil LogManager) to happen.

Reply:
What is ironic - my understanding that based on Google Coding style, in some codebases they prefer `int` over `size_t` for that exact reason... 🤷 the underflow....